src/NodeTypeResolver/PHPStan/TypeHasher.php

```diff
@@ -91,21 +91,7 @@ private function createUnionTypeHash(UnionType $unionType): string
             return $booleanType->describe(VerbosityLevel::precise());
         }
 
-        $normalizedUnionType = clone $unionType;
```

Comment:
the clone is somehow still needed, see error in test https://github.com/rectorphp/rector-src/actions/runs/8169922103/job/22334963648#step:5:93

```suggestion
        $clonedUnionType = clone $unionType;
        return $clonedUnionType->describe(VerbosityLevel::precise());
```